pkg/reconciler/revision/resources/queue.go

```diff
@@ -256,21 +256,41 @@ func makeQueueContainer(rev *v1.Revision, cfg *config.Config) (*corev1.Container
 	ports = append(ports, servingPort)
 
 	container := rev.Spec.GetContainer()
-	rp := container.ReadinessProbe.DeepCopy()
 
-	applyReadinessProbeDefaults(rp, userPort)
-
-	probeJSON, err := readiness.EncodeProbe(rp)
+	// During startup we want to poll the container faster than Kubernetes will
+	// allow, so we use an ExecProbe which starts immediately and then polls
+	// every 25ms. We encode the original probe as JSON in an environment
+	// variable for this probe to use.
+	userProbe := container.ReadinessProbe.DeepCopy()
+	applyReadinessProbeDefaultsForExec(userProbe, userPort)
+	execProbe := makeExecProbe(userProbe)
+	userProbeJSON, err := readiness.EncodeProbe(userProbe)
```

Comment:
I assume that using the readiness probe as the startup probe is consistent with K8s when the latter is unspecified?  Have you thought at all about letting users specify `startupProbe`?

Reply:
yah, as I understand it startupprobe just pauses the other probes until it succeeds (so we avoid accidentally probing before the pod is ready and ending up being fined a periodSeconds until we can try again). It doesn't have any "extra" semantics we'll be accidentally importing.

Re: user startupProbe yes I am! Will open an issue in a minute. They're less important, I think, for our use case because generally they're used for slow starting containers but there's no real reason _not_ to let users use them (especially since they should be wholly orthogonal if applied to user container rather than QP).